doc/examples/boilerplate.py

```diff
@@ -0,0 +1,96 @@
+"""
+A docstring
+
+This can double as a helpful message which explains how the program is run.
+"""
+
+from __future__ import absolute_import, division, print_function
+
+import logging
+import sys
+
+import libtbx.phil
+import dials.util
+import dials.util.log
+from dials.util.options import OptionParser
+from dials.util.options import flatten_reflections
+from dials.util.options import flatten_experiments
+
+
+# Define a logger
+logger = logging.getLogger("dials.logger_name")
+
+# Define the master PHIL scope for this program
+phil_scope = libtbx.phil.parse(
+    """
+output {
+    reflections = stronger.refl
+        .type = path
+    log = dials.command_name.log
+        .type = path
+}
+bool_parameter = False
+  .type = bool
+integer_parameter = 0
+  .type = int
+"""
+)
+
+
+def do_stuff(experiments, reflections, params):
+    """Write the behaviour of the program as functions and classes outside run()"""
+    logger.info("Hello world!")
+    # Include file output here
+    logger.info("integer_parameter: %i", params.integer_parameter)
+    logger.info("bool_parameter: %s", params.bool_parameter)
+    reflections.as_file(params.output.reflections)
```

Comment:
I have a concern about this.  As an API user, I often want to make use of an algorithm from a DIALS module, retain the reflection table/experiment list/whatever, do further calculations and only then write an output file.  I may never wish to produce the file output that the DIALS module would normally produce when called as a stand-alone program.  So insisting on packaging up file output in functions together with the reusable algorithm code makes things difficult.

Reply:
I appreciate that the corollary is that if one _does_ wish to produce the normal file output when re-using parts of a DIALS program via function calls, one has to put in a few `reflections.as_file(params.output.reflections)`-esque lines, but I think explicit is better than implicit...

Reply:
I hold no strong opinions here, but I note it was you who added the output lines to the boilerplate in the wiki 🙃
https://github.com/dials/dials/wiki/Boilerplate-(or-the-standard-framework-of-a-DIALS-program)/_compare/e63e7126920cebc0069ad65362488b57917bb729

This was just a straightforward copy of the existing wiki boilerplate, with minimal changes to ensure it was actually valid code. Happy for you to change this as you see fit.

Reply:
Yes, that was uploaded after some wrangling by committee.  I wasn't entirely satisfied with it at the time, but the whole thing was enough of an improvement on the previous boilerplate that it didn't seem worth splitting hairs.  I might push a change and pass the review baton back to you.  Perhaps we can get a couple of other pairs of eyes on it too, once we're both satisfied.  I don't feel 100% confident that I know what's best here.